src/main/java/com/tananaev/adblib/AdbConnection.java

```diff
@@ -285,8 +285,14 @@ public boolean connect(long timeout, TimeUnit unit) throws IOException, Interrup
 
         /* Wait for the connection to go live */
         synchronized (this) {
-            if (!connected)
-                wait(unit.toMillis(timeout));
+            long timeoutDurationMillis = unit.toMillis(timeout);
+            long timeoutEndMillis = System.currentTimeMillis() + timeoutDurationMillis;
```

Comment:
I think we can simplify it even more. What if we just leave `timeoutEndMillis`.

Reply:
I've just inlined `timeoutDurationMillis` in 8f54e7b. Is that what you mean?

Reply:
It actually looks good, but that's not what I meant. You don't actually need both `timeoutEndMillis` and `remainingTimeoutMillis`.

Reply:
Ah, I see what you mean. I've made the change in 61f4441.